### PR TITLE
feat: rename search props (ET-693)

### PR DESCRIPTION
**Pull request by jesse-amano-hpe**
## Ticket

[ET-693]

## Description

Spike + initial PR to begin working on the v2 API as proposed in the [Run-Centric API PRD](https://hpe-aiatscale.atlassian.net/wiki/spaces/DW/pages/1672937530/PRD+-+Run+Centric+API+CLI+SDK).

Several simplifications and shortcuts were attempted at first, but did not make it in:

The `additional_bindings` keyword does not serve our purpose here, as it (by design) funnels requests on different endpoint paths to the same handler, which _is_ useful, just not what we wanted this for. Doing this would have required us to make `search_id` and `experiment_id` two optional properties on a shared request object, and associate a handler func that just calls one of two different functions depending on which property isn't zero-valued. Ultimately, this ends up looking a lot like this same PR in terms of code bloat and complexity, but with the additional downside that we'd counterintuitively be heading toward defining dozens of "v2" structs under folders like `/api/v1` and making it harder to someday deprecate parts of the v1 API.

Proto marshal/unmarshal appears to be fast and pass tests _as long as we do not change any of the structure of the "new" definitions_ but this requirement makes it brittle even if it technically could have worked this one time, for this one thing. It seems better to handle mappings explicitly, in any case, and take the opportunity to deprecate what we can. (Unfortunately, I don't think we actually can fully remove the "config" property even though it has been marked as deprecated for a long time.)

I intentionally avoided duplicating the lower-level logic of actually fetching an experiment; the intent of this is to establish a pattern we'd like to repeat, in order to get through renaming all the endpoints and properties quickly, and I'm interested in finding ways to reduce the amount we are changing and would have to also duplicate testing and support efforts for -- and in receiving feedback on any of these shortcuts we _shouldn't_ be taking.

## Test Plan

Integration tests.

## Checklist

- [x] Changes have been manually QA'd
- [x] New features have been approved by the corresponding PM
- [x] User-facing API changes have the "User-facing API Change" label
- [x] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses have been included for new code which was copied and/or modified from any external code

[ET-693]: https://hpe-aiatscale.atlassian.net/browse/ET-693?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by AmanuelAaron — COMMENTED** [review]
Looks good so far. Should this be merging into main? Or would it be better to have a feature branch?

File: master/internal/api_search.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	"github.com/determined-ai/determined/proto/pkg/jobv1"
+	"github.com/determined-ai/determined/proto/pkg/jobv2"
+	"github.com/determined-ai/determined/proto/pkg/searchv2"
+)
+
+func convertExpToSearch(exp *experimentv1.Experiment) *searchv2.Search {
+	return &searchv2.Search{
+		Id:                    exp.Id,
+		Description:           exp.Description,
+		Labels:                exp.Labels,
```
Comment (AmanuelAaron) [inline master/internal/api_search.go:27]: We should actually rename these as Tags to match frontend

File: proto/src/determined/search/v2/search.proto [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+        "config"
+      ]
+    }
+  };
+  // The id of the search.
+  int32 id = 1;
+  // The description of the search.
+  string description = 2;
+  // Labels attached to the search.
+  repeated string labels = 3;
```
Comment (AmanuelAaron) [inline proto/src/determined/search/v2/search.proto:88]: update to tags here

**Review by AmanuelAaron — APPROVED** [review]

File: proto/src/determined/search/v2/search.proto [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+  // The id for the run associated with this validation entry.
+  int32 run_id = 1;
+  // The time at which the completed validation was reported.
+  google.protobuf.Timestamp end_time = 2;
+  // The value of the `searcher.metric`, indicated by the search config, for
+  // the validation.
+  float searcher_metric = 3;
+}
+
+// File node is one node of file in search model definition file tree.
```
Comment (corban-beaird) [inline proto/src/determined/search/v2/search.proto:225]: nit (super nit pick):
```suggestion
// File node is one node of a file in search model definition file tree.
```

**Review by corban-beaird — APPROVED** [review]
Looks solid!

**Pull request merged**